src/test/java/com/googleinterns/zoomtube/servlets/DiscussionServletTest.java

```diff
@@ -0,0 +1,205 @@
+  }
+
+  @After
+  public void cleanUp() {
+    testServices.tearDown();
+  }
+
+  @Test
+  public void doPost_loggedOut_postForbidden() throws ServletException, IOException {
+    testServices.setEnvIsLoggedIn(false);
+    when(request.getParameter(DiscussionServlet.PARAM_LECTURE)).thenReturn(LECTURE_ID_STR);
+
+    servlet.doPost(request, response);
+
+    verify(response).sendError(eq(HttpServletResponse.SC_FORBIDDEN), anyString());
+    assertThat(datastore.prepare(new Query(Comment.ENTITY_KIND)).countEntities(withLimit(1)))
+        .isEqualTo(0);
+  }
+
+  @Test
+  public void doPost_storesCommentWithProperties_noParent() throws ServletException, IOException {
+    testServices.setEnvIsLoggedIn(true);
+    testServices.setEnvEmail("author@example.com");
+    when(request.getParameter(DiscussionServlet.PARAM_LECTURE)).thenReturn(LECTURE_ID_STR);
+    when(request.getReader()).thenReturn(new BufferedReader(new StringReader("Something unique")));
+
+    servlet.doPost(request, response);
+
+    verify(response).setStatus(HttpServletResponse.SC_ACCEPTED);
+    PreparedQuery query = datastore.prepare(new Query(Comment.ENTITY_KIND));
+    assertThat(query.countEntities(withLimit(2))).isEqualTo(1);
+    Comment comment = Comment.fromEntity(query.asSingleEntity());
+    assertThat(comment.lecture().getId()).isEqualTo(LECTURE_ID);
+    assertThat(comment.author().getEmail()).isEqualTo("author@example.com");
+    assertThat(comment.content()).isEqualTo("Something unique");
+    assertThat(comment.parent().isPresent()).isFalse();
+  }
+
+  @Test
+  public void doPost_storesCommentParent() throws ServletException, IOException {
+    final int parentId = 32;
+    testServices.setEnvIsLoggedIn(true);
+    when(request.getParameter(DiscussionServlet.PARAM_LECTURE)).thenReturn(LECTURE_ID_STR);
+    when(request.getParameter(DiscussionServlet.PARAM_PARENT))
+        .thenReturn(Integer.toString(parentId));
+    when(request.getReader())
+        .thenReturn(new BufferedReader(new StringReader(TEST_COMMENT_CONTENT)));
+
+    servlet.doPost(request, response);
+
+    verify(response).setStatus(HttpServletResponse.SC_ACCEPTED);
+    PreparedQuery query = datastore.prepare(new Query(Comment.ENTITY_KIND));
+    Comment comment = Comment.fromEntity(query.asSingleEntity());
+    assertThat(comment.parent().isPresent()).isTrue();
+    assertThat(comment.parent().get().getId()).isEqualTo(parentId);
+  }
+
+  @Test
+  public void doGet_returnsNothingForUnknownLecture() throws ServletException, IOException {
+    when(request.getParameter(DiscussionServlet.PARAM_LECTURE)).thenReturn(LECTURE_ID_STR);
+    StringWriter content = new StringWriter();
+    PrintWriter writer = new PrintWriter(content);
+    when(response.getWriter()).thenReturn(writer);
+
+    servlet.doGet(request, response);
+
+    verify(response).setContentType("application/json;");
+    List<Comment> comments = getCommentsFromJson(content.toString());
+    assertThat(comments.isEmpty()).isTrue();
```

Comment:
you can rewrite it as: assertThat(comments).isEmpty();

Reply:
Not supported by this version of truth